File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.Arrays;
class Solution {

	public static int[] solution(int[] arr, int[][] queries) {
		int[] answer = new int[queries.length];
		
		

		for (int j = 0; j < queries.length; j++) {
			answer[j]=Integer.MAX_VALUE;

			for (int i = queries[j][0]; i < queries[j][1] + 1; i++) {
				
				if (queries[j][2] < arr[i]) {
					
					answer[j]=answer[j]>arr[i]?arr[i]:answer[j];

			}
			
			// i에 해당하는값들을 arr에서 가져온다음에 k보다 큰 값을 변수에 저장해가면서 최솟값을 저장해나가는 방식은 안되는가요?

			
			
			
			
			
			
		}
			answer[j] = answer[j] == Integer.MAX_VALUE ? -1 : answer[j];}
		
		return answer;
	
	}
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
  return queries.map((el) => {
    arrtmp = [...arr]
      .filter((e, i) => el[0] <= i && i <= el[1] && e > el[2])
      .sort((a, b) => a - b);
    return !arrtmp.length ? -1 : arrtmp[0];
  });
}
